File: DotNetMessaging/backend/DotNetMessaging.API/Models/MessageRead.cs
using MongoDB.Bson.Serialization.Attributes;

namespace DotNetMessaging.API.Models;

public class MessageRead : BaseEntity
{
    [BsonElement("messageId")]
    public string MessageId { get; set; } = string.Empty;

    [BsonElement("userId")]
    public string UserId { get; set; } = string.Empty;

    [BsonElement("readAt")]
    public DateTime ReadAt { get; set; } = DateTime.UtcNow;
}

File: DotNetMessaging/SpringBootMessaging/src/main/java/com/messaging/api/models/MessageRead.java
package com.messaging.api.models;

import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Data
@EqualsAndHashCode(callSuper = true)
@Document(collection = "messageReads")
public class MessageRead extends BaseEntity {
    private String messageId;
    
    private String userId;
    
    private LocalDateTime readAt = LocalDateTime.now();
}

